services/registry/test/models/user.js

```diff
@@ -0,0 +1,173 @@
+        await addNewUser('foo bar', 'bar@foo.com');
+      } catch (e) {
+        error = e.message;
+      }
+
+      demand(error).to.be('Usernames must be unique.');
+    })
+  );
+
+  it(
+    'throws an error when usernames are not at least 1 character',
+    providePostgres(async () => {
+      let error = undefined;
+
+      try {
+        await addNewUser('', 'bar@foo.com');
+      } catch (e) {
+        error = e.message;
+      }
+
+      demand(error).to.be(USER_ERRORS.NO_NAME);
+    })
+  );
+
+  describe('signup', () => {
+    it(
+      'can sign a user up with valid parameters',
+      providePostgres(async () => {
+        const newSignup = await User.signup('foo', 'bar@foo.com', false);
+
+        demand(newSignup instanceof User).to.be.truthy();
+        demand(newSignup.active).to.be.truthy();
+      })
+    );
+
+    it(
+      'throws an error given an invalid name',
+      providePostgres(async () => {
+        let error = undefined;
+        try {
+          const u = await User.signup('', 'bar@foo.com', false);
+          console.log(u);
+        } catch (e) {
+          error = e.message;
+        }
+
+        demand(error).to.be(USER_ERRORS.NO_NAME);
+      })
+    );
+
+    it(
+      'creates a namespace for the user',
+      providePostgres(async () => {
+        const name = 'foo bar';
+        const newUser = await User.signup(name, 'bar@foo.com', false);
+
+        const userNamespace = await Namespace.objects.get({
+          active: true,
+          name: newUser.name
+        });
+
+        demand(userNamespace instanceof Namespace).to.be.truthy();
+        demand(userNamespace.active).to.be.truthy();
+      })
+    );
+
+    it(
+      'creates a namespace with a host for the user',
+      providePostgres(async () => {
+        const name = 'foo bar';
+        await User.signup(name, 'bar@foo.com', false);
+
+        // Fetch the user's namespace that was created during signup.
+        const userNamespace = await Namespace.objects.filter({
+          'host.id': 1, // TODO: don't hardcode?
+          name
+        });
+
+        // User should only have 1 active namespace
+        demand(userNamespace.length).to.be(1);
+        demand(userNamespace[0].active).to.be.truthy();
+      })
+    );
+
+    it(
+      'should not create a remoteAuth when remoteAuth param is false',
+      providePostgres(async () => {
+        const name = 'foo bar';
+        const user = await User.signup(name, 'bar@foo.com', false);
+
+        const remoteAuthObjects = await Authentication.objects.filter({
+          user
+        });
+
+        // No authentication objects should exist for this user
+        demand(remoteAuthObjects.length).to.be(0);
+      })
+    );
+
+    it(
+      'should create a remoteAuth when remoteAuth param is provided',
+      providePostgres(async () => {
+        // Kind of dirty, but we currently only need it for this test
+        let prevOAuthEnvVar = process.env.OAUTH_PASSWORD;
```

Comment:
Like my comment says, it's not the best. Altering `process.env` feels weird ... but I think we would need some sort of global setup script that would load something like a `.env-test` file to avoid this

Reply:
Better to use the .env-test file for this.

Reply:
I don't mind messing with the environment here. I might suggest doing this bit in a separate "before / after" pair to make sure it's always run.

Reply:
> I might suggest doing this bit in a separate "before / after" pair to make sure it's always run.
Done!